infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactory.java

```diff
@@ -30,19 +47,106 @@
  */
 @Singleton
 public class OpenShiftProjectFactory extends KubernetesNamespaceFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(OpenShiftProjectFactory.class);
 
   private final OpenShiftClientFactory clientFactory;
+  private final String defaultNamespaceName;
+  private final boolean allowUserDefinedNamespaces;
 
   @Inject
   public OpenShiftProjectFactory(
       @Nullable @Named("che.infra.openshift.project") String projectName,
       @Nullable @Named("che.infra.kubernetes.service_account_name") String serviceAccountName,
       @Nullable @Named("che.infra.kubernetes.cluster_role_name") String clusterRoleName,
-      OpenShiftClientFactory clientFactory) {
-    super(projectName, serviceAccountName, clusterRoleName, clientFactory);
+      @Nullable @Named("che.infra.kubernetes.namespace.default") String defaultNamespaceName,
+      @Named("che.infra.kubernetes.namespace.allow_user_defined")
+          boolean allowUserDefinedNamespaces,
+      OpenShiftClientFactory clientFactory,
+      OpenShiftClientConfigFactory clientConfigFactory) {
+    super(
+        projectName,
+        serviceAccountName,
+        clusterRoleName,
+        defaultNamespaceName,
+        allowUserDefinedNamespaces,
+        clientFactory);
+    if (allowUserDefinedNamespaces && !clientConfigFactory.isPersonalized()) {
+      LOG.warn(
+          "Users allowed to list projects but all reuse Che Server configured service account. "
```

Comment:
Makes sense, but is a little bit difficult to parse for me personally. What about:
`Users are allowed to list projects but Che server is configured with a service account. All users will receive the same list of projects. Consider ...`

Reply:
The wording is not my strong side )
Applied proposed variant